### PR TITLE
[RISCV][MC] Add support for experimental Zcmop extension

### PR DESCRIPTION
**Pull request by wangpc-pp**
This implements experimental support for the Zcmop extension as
specified here: https://github.com/riscv/riscv-isa-manual/blob/main/src/zimop.adoc.

This change adds only MC support.

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVFeatures.td [hunk 1]
```diff
@@ -693,6 +693,13 @@ def HasStdExtZimop : Predicate<"Subtarget->hasStdExtZimop()">,
                                AssemblerPredicate<(all_of FeatureStdExtZimop),
                                "'Zimop' (May-Be-Operations)">;
 
+def FeatureStdExtZcmop : SubtargetFeature<"experimental-zcmop", "HasStdExtZcmop", "true",
+                                          "'Zcmop' (Compressed May-Be-Operations)",
+                                          [FeatureStdExtZca]>;
```
[the remark this answers is not in the document]
Comment (topperc) [reply llvm/lib/Target/RISCV/RISCVFeatures.td:698]: I think so. The specs have been very inconsistent on terminology. The fundamental issue is that the specs define hardware and don't think about how `-march` in tools works. What the spec is saying is that if your hardware supports Zcmop, it must support Zca.

Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVFeatures.td:698]: Any features implications in RISCV.td must also exist in RISCVISAInfo.cpp.

File: llvm/lib/Target/RISCV/RISCVInstrInfoZcmop.td [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+class CMOPInst<bits<3> imm3, string opcodestr>
+    : RVInst16CI<0b011, 0b01, (outs), (ins), opcodestr, ""> {
+  let Inst{6-2} = 0;
+  let Inst{7} = 1;
+  let Inst{10-8} = imm3;
+  let Inst{12-11} = 0;
+}
+
+foreach i = 0...7 in {
+    let Predicates = [HasStdExtZcmop, HasStdExtCOrZca] in {
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVInstrInfoZcmop.td:26]: Reduce indentation to 2 spaces

File: llvm/lib/Target/RISCV/RISCVFeatures.td (hunk 1 above)
[the remark this answers is not in the document]
Comment (kito-cheng) [reply llvm/lib/Target/RISCV/RISCVFeatures.td:698]: My understanding for those terminology on RISC-V ISA spec is: imply == depend == require

See comment from Krste and Andrew in this issue: https://github.com/riscv/riscv-v-spec/issues/723#issuecomment-922153867

**Review by yetingk — APPROVED** [review]
LGTM. But please wait for @topperc's review.

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**